File: plugin/download/ios/Classes/service/DownloadService.swift
//
//  DownloadService.swift
//  Runner
//
//  Created by SamLS Chen on 2024/8/19.
//

import Foundation

protocol DownloadService{
    
    func createDownload(urlString: String, isConcurrent: Bool) async -> ServiceResult<Int>
    func startDownload(downloadEntity: DownloadEntity) async
    func resumeDownload(downloadID: String) async -> ServiceResult<Int>
    func cancelDownload(downloadID: String) async -> ServiceResult<Int>
    func pauseDownload(downloadID: String) async -> ServiceResult<Int>
    func manualPauseDownload(downloadID: String) async -> ServiceResult<Int>
}

File: plugin/download/android/src/main/kotlin/com/example/perfect_corp_homework/download/service/DownloadService.kt
package com.example.perfect_corp_homework.download.service

import com.example.perfect_corp_homework.download.api.ServiceResult
import com.example.perfect_corp_homework.download.model.DownloadEntity
import kotlinx.coroutines.CoroutineScope

interface DownloadService: CoroutineScope{
    suspend fun createDownload(urlString: String, isConcurrent: Boolean): ServiceResult<Int>
    suspend fun startDownload(downloadEntity: DownloadEntity): Unit
     fun pauseDownload(downloadID: String): ServiceResult<Int>
     suspend fun resumeDownload(downloadID: String): ServiceResult<Int>
     fun cancelDownload(downloadID: String): ServiceResult<Int>
     fun manualPauseDownload(downloadID: String): ServiceResult<Int>
}